client/data/account-keys/hooks.js

```diff
@@ -0,0 +1,80 @@
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useCallback } from 'react';
+import { STORE_NAME } from '../constants';
+
+export const useAccountKeys = () => {
+	const { saveAccountKeys } = useDispatch( STORE_NAME );
+
+	const accountKeys = useSelect( ( select ) => {
+		const { getAccountKeys } = select( STORE_NAME );
+
+		return getAccountKeys();
+	}, [] );
+
+	const isLoading = useSelect( ( select ) => {
+		const { hasFinishedResolution, isResolving } = select( STORE_NAME );
+
+		return (
+			isResolving( 'getAccountKeys' ) ||
+			! hasFinishedResolution( 'getAccountKeys' )
+		);
+	}, [] );
+
+	const isSaving = useSelect( ( select ) => {
+		const { isSavingAccountKeys } = select( STORE_NAME );
+
+		return isSavingAccountKeys();
+	}, [] );
+
+	return { accountKeys, isLoading, isSaving, saveAccountKeys };
+};
+
+export const useGetSavingError = () => {
+	return useSelect( ( select ) => {
+		const { getSavingError } = select( STORE_NAME );
+
+		return getSavingError();
+	}, [] );
+};
+
+const makeAccountKeysValueHook = ( attribute ) => () => {
```

Comment:
Nice approach :100:

Reply:
Thanks! This is kinda temporary and could still change if needed :) it's not set in stone, I just put it there as an example - if people need to change it, feel free to do so! :)